src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuery.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.bigquery.stepsdesign;
+
+import io.cdap.e2e.pages.actions.CdfBigQueryPropertiesActions;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.pages.locators.CdfBigQueryPropertiesLocators;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.ConstantsUtil;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.common.stepsdesign.TestSetupHooks;
+import io.cdap.plugin.utils.E2ETestConstants;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.time.DateUtils;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * BigQuery related stepDesigns.
+ */
+public class BigQuery implements CdfHelper {
+
+  static {
```

Comment:
Looks like BQ source and sink step designs are in the same class. I think it would be more clear if we had separate classes for BQ source and sink because the plugin configurations are different. If there are configs that are common to source and sink (ex - project id), we could put that in a base class.
WDYT @itsankit-google @tivv ?

Reply:
+1. I think the structure of step definitions also can be similar to structure of features, i.e. `stepsdesign/bigquery/BigQuerySource.java`. Can add a `BigQueryBase` class for common step defintions. `BigQuerySource` and `BigQuerySink` to have specific step definitions. Similar approach can be taken for GCS as well.

Reply:
I don't think we need to have separate step definition files.  We are already indicating in stepdef and method name if property is sink or source specific.
for eg: 
`@Then("Enter BigQuery source property filter {string}")`
`@Then("Toggle BigQuery sink property truncateTable to true")`

Reply:
Created separate classes for source and sink step definitions.